src/main/java/com/google/roomies/data/Listing.java

```diff
@@ -0,0 +1,201 @@
+    abstract Builder setDocumentId(Optional<String> documentId);
+    abstract Builder setTitle(String title);
+    abstract Builder setDescription(String description);
+    abstract Builder setStartDate(Date startDate);
+    abstract Builder setEndDate(Date endDate);
+    abstract Builder setLeaseType(LeaseType leaseType);
+    abstract Builder setNumRooms(int numRooms);
+    abstract Builder setNumBathrooms(int numBathrooms);
+    abstract Builder setNumShared(int numShared);
+    abstract Builder setNumSingles(int numSingles);
+    abstract Builder setSharedPrice(Money sharedPrice);
+    abstract Builder setSinglePrice(Money singlePrice);
+    abstract Builder setListingPrice(Money listingPrice);
+
+    abstract LeaseType leaseType();
+    abstract int numRooms();
+
+    abstract Listing build();
+    
+    /**
+    * Throws IllegalArgumentException if input does not match a LeaseType enum 
+    * value (case sensistive).
+    */
+    public Builder setLeaseType(String leaseType) {
+      setLeaseType(LeaseType.valueOf(leaseType));
+      return this;
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumRooms(String numRooms) {
+      setNumRooms(Integer.parseInt(numRooms));
+      return this;
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumBathrooms(String numBathrooms) {
+      setNumBathrooms(Integer.parseInt(numBathrooms));
+      return this;     
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumShared(String numShared) {
+      setNumShared(Integer.parseInt(numShared));
+      return this;     
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumSingles(String numSingles) {
+      setNumSingles(Integer.parseInt(numSingles));
+      return this;     
+    }
+
+    public Builder setStartDate(String startDate) throws ParseException {
+      setStartDate(StringConverterUtils.stringToDate(startDate));
+      return this;
+    }
+
+    public Builder setEndDate(String endDate) throws ParseException {
+      setEndDate(StringConverterUtils.stringToDate(endDate));
+      return this;
+    }
+
+    public Builder setSharedPrice(String sharedPrice) {
+      setSharedPrice(StringConverterUtils.stringToMoney(sharedPrice));
+      return this;
+    }
+
+    public Builder setSinglePrice(String singlePrice) {
+      setSinglePrice(StringConverterUtils.stringToMoney(singlePrice));
+      return this;
+    }
+
+    public Builder setListingPrice(String listingPrice) {
+      setListingPrice(StringConverterUtils.stringToMoney(listingPrice));
+      return this;
+    }
+
+    /**
+    * Sets all listing values to the corresponding HTTP Servlet request parameter.
+    */
+    public Builder fromServletRequest(HttpServletRequest request) throws ParseException {
```

Comment:
Should this be a method on the builder, or the autovalue itself? In usuage you call fromServeltRequest().build(), so you might as well just return the fully-formed object.